### DISCUSSION TITLE
Share individual image from mobile browser?

### DISCUSSION THREAD

**Discussion by rmzg** [photoprism/photoprism#3494] [General]
Right now on desktop you can use rightclick/context menus to effectively share a single image via its permanent url, which is fine, but it'd be great to have that functionality available on mobile browsers, which certainly make it difficult to do basic operations like get the url of an image, especially when you have invisible divs overlayed on top of it to create UI elements.

I'm aware of sharing albums but it's a whole lot of clicks, especially on mobile, to share a single image. 

Just as a quick option you could add it to the "context menu" that you get in the bottom right when you select an image in the mass view mode, or just add a link button in the "View image" ui when you touch a single image. Either would be great.

**Comment by graciousgrey** [comment]
Sharing of individual files in a similar way as albums can be shared is planned:

- https://github.com/photoprism/photoprism/issues/466

What is already possible on mobile is copying/sharing photos using the long touch menu.

If you are in a mobile browser on iOS, you can long touch the image in full screen mode to copy/share or save the photo 
- https://github.com/photoprism/photoprism/issues/1233

Unfortunately, this does not yet work for the PWA 
- https://github.com/photoprism/photoprism/issues/3362).

On Android, the long-touch menu is available both in the PWA and directly in the browser.

**Comment by rmzg** [comment]
You are completely correct, you do have long press options on the image in
question, although android/chrome does make it remarkably annoying to get
the actual url for sharing.

Where I was actually having the problem I described above was with
*videos*. On a desktop you can get the url and share that but I haven't
figured out a mobile method, and the various built-in "sharing" options are
considerably less useful when you have a 50+ mb video. Getting the url of a
video to share would be hugely helpful.

**Reply by graciousgrey** [reply]
You are right the long touch menu does not work well for videos. #466 will be implemented in a way that works for all file types.
